### PR TITLE
Workers and Buildings

### PR DESCRIPTION
**Pull request by ZakCollins**
Added models for the worker, base and barracks. I couldn't really find exactly what I had in mind for any of them but they look decent. Fixed some bugs with the menus and optimized some code.

### CODE REVIEW

File: Assets/2 Scripts/RTSObjects/Unit/Unit.cs [hunk 1]
```diff
@@ -58,15 +58,8 @@ public abstract class Unit : RTSObject {
...
-		healthIndicator.fillAmount = currentHealth / 100f;
-		if (currentHealth <= 0) {
-			//Probably play some animation first
-			if (selected) {
-				OnUnselected();
-			}
-			Destroy (gameObject);
-		}
+		base.TakeDamage(amount);
+		healthIndicator.fillAmount = currentHealth / (float)maxHealth;
```
Comment (thejinxters) [inline Assets/2 Scripts/RTSObjects/Unit/Unit.cs]: Thanks for fixing this. I am not sure what I was thinking with the 100f thing. :smiley_cat:

File: Assets/2 Scripts/RTSObjects/Building/Building.cs [hunk 2]
```diff
@@ -1,16 +1,30 @@
...
+
+
 	//*** Public Functions
 
 	public void BuildUnit(Unit unit) {
 
 	}
 
+	public override void TakeDamage(int amount) {
+		base.TakeDamage(amount);
```
Comment (thejinxters) [inline Assets/2 Scripts/RTSObjects/Building/Building.cs]: I assume base.TakeDamage is sort of like calling "super" in java?

File: Assets/2 Scripts/CameraController.cs [hunk 3]
```diff
@@ -4,8 +4,8 @@
 public class CameraController : MonoBehaviour {
 
 	//Inspector Variables
-	public Camera camera;
-	public float cameraSpeed;
+	private Camera camera;
+	private float cameraSpeed;
```
Comment (thejinxters) [inline Assets/2 Scripts/CameraController.cs]: Yay! private variables, although we should move them into the private variable section (in the comments) to keep things neat.

**Comment by thejinxters** [comment]
All the changes you made look good in the code. When I opened the project, the barracks and worker did not load its model properly. Home base loaded just fine. Any ideas what may cause that? I had a blender error (blender not installed) when I first opened the project, so I installed blender. However, the models are still invisible.

**Comment by thejinxters** [comment]
Nevermind. I checked out to a different branch, loaded unity, then rechecked out to the worker branch (After blender was installed) and it works great! 

I give this PR :+1:

File: Assets/2 Scripts/RTSObjects/Building/Building.cs (hunk 2 above)
Comment (ZakCollins) [inline Assets/2 Scripts/RTSObjects/Building/Building.cs]: Yep, thats exactly what base is.

**Comment by ZakCollins** [comment]
Oh yea, I'm pretty sure you need blender installed in order to import blender models with Unity. I forgot about that. I also changed the MenuSystem back to the way it was. I feel like it is much more flexible this way than having a bool for each possible menu system and then doing a find (which is really slow), but I just commented out the old way so we can change it back if we want.

**Comment by thejinxters** [comment]
@ZakCollins - I saw that (excellent), I think you are good to merge!

**Pull request merged**